example/spec/parallel/ingestGranule/IngestGranuleSuccessSpec.js

```diff
@@ -160,11 +156,14 @@ describe('The S3 Ingest Granules workflow', () => {
     await Promise.all(inputPayload.granules[0].files.map((fileToTag) =>
       s3().putObjectTagging({ Bucket: config.bucket, Key: `${fileToTag.path}/${fileToTag.name}`, Tagging: { TagSet: expectedS3TagSet } }).promise()));
 
+    const collectionUrlString = '{cmrMetadata.Granule.Collection.ShortName}___{cmrMetadata.Granule.Collection.VersionId}/{substring(file.name, 0, 3)}/';
     expectedSyncGranulePayload = loadFileWithUpdatedGranuleIdPathAndCollection(templatedSyncGranuleFilename, granuleId, testDataFolder, newCollectionId);
     expectedSyncGranulePayload.granules[0].dataType += testSuffix;
+    expectedSyncGranulePayload.granules[0].files = addUrlPathToGranuleFiles(expectedSyncGranulePayload.granules[0].files, testId, '');
     expectedPayload = loadFileWithUpdatedGranuleIdPathAndCollection(templatedOutputPayloadFilename, granuleId, testDataFolder, newCollectionId);
     expectedPayload.granules[0].dataType += testSuffix;
-
+    expectedPayload.granules = addUniqueGranuleFilePathToGranuleFiles(expectedPayload.granules, testId);
+    expectedPayload.granules[0].files = addUrlPathToGranuleFiles(expectedPayload.granules[0].files, testId, collectionUrlString);
```

Comment:
I am definitely missing something, but why doesn't `addUrlPathToGranuleFiles` run for the files of all granules? If it did, then these two helpers could be combined?

Reply:
addUniqueGranuleFilePathToGranuleFiles updates the filename/filepath.    In retrospect, this should be named granule*s* files.  

addUrlPathToGranuleFiles updates the url_paths for a granule object.   This is used to update File objects with the updated url_path, and is required for payload matching.  

For example, we're not adding granule filepath  nearl L162 above to the fixture because SyncGranules shouldn't have the filepaths yet.